src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -0,0 +1,478 @@
+
+  def resolve(self, targets, compile_classpath):
+    """
+
+    :param targets: a collection of targets to do 3rdparty resolve against
+    :param compile_classpath: classpath product that holds the resolution result. IMPORTANT: this parameter will be changed.
+    :return:
+    """
+
+    coursier_subsystem_instance = CoursierSubsystem.global_instance()
+
+    coursier_jar = coursier_subsystem_instance.bootstrap_coursier()
+
+    manager = JarDependencyManagement.global_instance()
+
+    jar_targets = manager.targets_by_artifact_set(targets)
+
+    for artifact_set, target_subset in jar_targets.items():
+      # TODO(wisechengyi): this is the only place we are using IvyUtil method, which isn't specific to ivy really.
+      raw_jar_deps, global_excludes = IvyUtils.calculate_classpath(target_subset)
+
+      with self.invalidated(target_subset,
+                            invalidate_dependents=False,
+                            silent=False,
+                            fingerprint_strategy=CouriserResolveFingerprintStrategy([])) as invalidation_check:
+
+        target_resolution_filename = 'coursier_resolve.json'
+        if not invalidation_check.invalid_vts:
+          success = self._load_result_from_cache(compile_classpath, invalidation_check.all_vts,
+                                               target_resolution_filename)
+          if success:
+            return
+
+        jars_to_resolve, local_exclude_args, pinned_coords = self._compute_jars_to_resolve_and_to_exclude(raw_jar_deps,
+                                                                                                          artifact_set,
+                                                                                                          manager)
+        # Prepare coursier args
+        coursier_cache_path = os.path.join(self.get_options().pants_bootstrapdir, 'coursier')
+        pants_jar_path_base = os.path.join(self.get_options().pants_workdir, 'coursier')
+        safe_mkdir(pants_jar_path_base)
+
+        coursier_workdir = os.path.join(self.get_options().pants_workdir, 'tmp')
+        safe_mkdir(coursier_workdir)
+
+
+        with temporary_file(coursier_workdir, cleanup=False) as f:
+          output_fn = f.name
+
+        common_args = ['fetch',
+                       # Print the resolution tree
+                       '-t',
+                       '--cache', coursier_cache_path,
+                       '--json-output-file', output_fn] + coursier_subsystem_instance.get_options().fetch_options
+
+        def construct_classifier_to_jar(jars):
+          product = defaultdict(list)
+          for x in jars:
+            product[x.coordinate.classifier or ''].append(x)
+          return product
+
+        classifier_to_jars = construct_classifier_to_jar(jars_to_resolve)
+
+        # Coursier calls need to be divided by classifier because coursier treats classifier option globally.
+        for classifier, classified_jars in classifier_to_jars.items():
```

Comment:
How well does this work with dependency graphs that cross classifiers?

Reply:
it is unclear right now. we do not have a specific to test. but anecdotally, 
if orgA:nameA:versionB:default depends on orgA:nameA:versionB:classifier, I think it should work.